File: packages/sdk-python/bidstack_crm/resources/custom_fields.py
"""Custom fields resource."""
from __future__ import annotations
from typing import Any
from ._base import BaseResource


class CustomFieldsResource(BaseResource):
    def list(self, entity_type: str) -> dict[str, Any]:
        """List custom field definitions for a given entity type."""
        return self._get("/custom-fields", params={"entityType": entity_type})

    def create(self, data: dict[str, Any]) -> dict[str, Any]:
        return self._post("/custom-fields", json=data)

    def update(self, field_id: str, data: dict[str, Any]) -> dict[str, Any]:
        return self._patch(f"/custom-fields/{field_id}", json=data)

    def delete(self, field_id: str) -> None:
        self._delete(f"/custom-fields/{field_id}")

File: packages/sdk-ruby/lib/bidstack_crm/resources/custom_fields.rb
require_relative "base"

module BidstackCrm
  module Resources
    class CustomFields < Base
      def list(entity_type) = get("/custom-fields", params: { entityType: entity_type })
      def create(data) = post("/custom-fields", body: data)
      def update(field_id, data) = patch("/custom-fields/#{field_id}", body: data)
      def destroy(field_id) = delete("/custom-fields/#{field_id}")
    end
  end
end
